### PR TITLE
feat: add support for accepting an Artifact Registry URL in pipeline_job

### PR DESCRIPTION
**Pull request by chongyouquan**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/python-aiplatform/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes #<issue_number_goes_here> 🦕

### CODE REVIEW

File: google/cloud/aiplatform/pipeline_jobs.py [hunk 1]
```diff
@@ -31,6 +31,7 @@
 
 from google.cloud.aiplatform.compat.types import (
     pipeline_job as gca_pipeline_job,
+    pipeline_job_v1beta1 as gca_pipeline_job_v1beta1,
```
Comment (sararob) [inline google/cloud/aiplatform/pipeline_jobs.py:34]: Let's wait until https://github.com/googleapis/python-aiplatform/pull/1221 is merged so you can change this to use `v1`.

File: google/cloud/aiplatform/pipeline_jobs.py [hunk 2]
```diff
@@ -223,15 +224,25 @@ def __init__(
...
+            "labels": labels,
+            "encryption_spec": initializer.global_config.get_encryption_spec(
                 encryption_spec_key_name=encryption_spec_key_name
             ),
-        )
+        }
+
+        if _VALID_AR_URL.match(template_path):
+            gca_pipeline_job_compat = gca_pipeline_job_v1beta1
+            pipeline_job_args["template_uri"] = template_path
```
Comment (sararob) [inline google/cloud/aiplatform/pipeline_jobs.py:239]: Could you update the docstring for `template_path` in the constructor to include info on AR uris?

File: google/cloud/aiplatform/pipeline_jobs.py [hunk 3]
```diff
@@ -299,7 +310,10 @@ def submit(
 
         _LOGGER.log_create_with_lro(self.__class__)
 
-        self._gca_resource = self.api_client.create_pipeline_job(
+        api_client = self.api_client
+        if hasattr(self._gca_resource, "template_uri"):
```
Comment (sararob) [inline google/cloud/aiplatform/pipeline_jobs.py:314]: Note to remove this when v1 support is available.

**Review by sararob — CHANGES_REQUESTED** [review]

File: google/cloud/aiplatform/pipeline_jobs.py (hunk 2 above)
Comment (chongyouquan) [reply google/cloud/aiplatform/pipeline_jobs.py:241]: Done.

File: google/cloud/aiplatform/pipeline_jobs.py (hunk 3 above)
Comment (chongyouquan) [reply google/cloud/aiplatform/pipeline_jobs.py]: Done.

File: google/cloud/aiplatform/pipeline_jobs.py (hunk 1 above)
Comment (chongyouquan) [reply google/cloud/aiplatform/pipeline_jobs.py]: Done.

**Review by sararob — CHANGES_REQUESTED** [review]
This looks good, thanks for making those changes! Last few things before it can be merged:
* Using `urllib` instead of `requests`
* KFP support in AR being publicly available

If I have any additional feedback as I try out the AR integration I'll add it here.

File: google/cloud/aiplatform/utils/yaml_utils.py [hunk 4]
```diff
@@ -95,3 +114,36 @@ def _load_yaml_from_local_file(file_path: str) -> Dict[str, Any]:
...
+        raise ImportError(
+            "pyyaml is not installed and is required to parse PipelineJob or PipelineSpec files. "
+            'Please install the SDK using "pip install google-cloud-aiplatform[pipelines]"'
+        )
+    auth = None
+    if credentials:
+        if not credentials.valid:
+            credentials.refresh(transport.requests.Request())
+        auth = ApiAuth(credentials.token)
+    response = requests.get(url=uri, auth=auth)
```
[the remark this answers is not in the document]
Comment (chongyouquan) [reply google/cloud/aiplatform/utils/yaml_utils.py]: Ok done, used `urllib.request` instead of `requests`.

**Comment by chongyouquan** [comment]
> This looks good, thanks for making those changes! Last few things before it can be merged:
> 
> * Using `urllib` instead of `requests`
> * KFP support in AR being publicly available
> 
> If I have any additional feedback as I try out the AR integration I'll add it here.

The use of `urllib` instead of `requests` is done.
For support in AR python library for KFP packages, it'll probably only be available later (in a few months' time I think).

**Review by sararob — APPROVED** [review]

**Review by parthea — APPROVED** [review]

**Pull request merged**